### DISCUSSION TITLE
Menu cursor shows during gameplay and plays cursor sound

### DISCUSSION THREAD

**Discussion by Walavouchey** [ppy/osu#20861] [Q&A]
just updated and this was the first thing i noticed.

https://user-images.githubusercontent.com/36758269/197352217-8605f7f6-461d-42c4-b4da-a9c9dc2d1713.mp4

frame rates due to obs. lazer on a surface pro 8 with touch input actually has very low input-to-hitsound latency for me nowadays which is very cool ty peppy

[runtime.log](https://github.com/ppy/osu/files/9844846/runtime.log)

version 2022.1022.0-lazer

**Comment by Bruno5430** [comment]
Related to #20713 ?

**Comment by nagi-desuuu** [comment]
Did you have the "show gameplay cursor during touch input" enabled, just in case?

**Reply by Walavouchey** [reply]
tested with both, but this is the menu cursor and not the gameplay cursor anyway

**Comment by peppy** [comment]
Does the cursor also show in the menus for you? It should on touch devices, so if that's the case then that will be the point where a fix needs to be applied.

**Reply by meowaphina** [reply]
> Does the cursor also show in the menus for you? It should on touch devices, so if that's the case then that will be the point where a fix needs to be applied.

It most likely is, though I think it could also be due to incorrect handling of cursors with td.

I can look into this later.

**Reply by peppy** [reply]
@WyvernX-YT please stop replying or I will have to limit your ability to reply. I understand you may think you are being helpful, but what you are currently posting across multiple threads is spam to us.

I am specifically asking the OP to reply to this to give more context, after actually testing what I asked for. Not blindly agreeing that the hypothesis is correct.

**Reply by Walavouchey** [reply]
it's basically a laptop with a touch screen. the menu cursor disappears if i drag around, but appears if i just tap. same for the gameplay cursor, i.e. in the osu! ruleset

using the trackpad makes it appear like normal, as does enabling "show gameplay cursor during touch input" (even though it affects both cursors)

https://user-images.githubusercontent.com/36758269/197496196-29ddaca0-f745-426a-abed-1dc3f1fd09af.mp4

also as an edge case tidbit (not shown in the video), with the setting turned off, using a pen (against the touch screen) will show the cursor when hovering and hide the cursor while dragging, which i think is quite neat

**Comment by Walavouchey** [comment]
i want to gently bump this. it's been fun slowly getting into mania with the new gameplay scale/rotation changes but this is still quite the annoyance when playing with effect sounds enabled

**Reply by peppy** [reply]
Hmm, I guess this only occurs on desktop touch devices correct? Since the cursor is hidden on mobile (correctly for me on iOS, at least)

**Reply by Qerewe** [reply]
This also applies to android. Playing mania/ctb/taiko with touch while having "show gameplay cursor during touch input" makes the mouse appear and a noise whenever you tap. The mouse appearing is intended with that setting I'm guessing but the cursor sound is distracting

**Reply by peppy** [reply]
I don't think that setting should cause the menu cursor to appear. In my eyes it should only affect *actual* gameplay cursors, ie the osu! ruleset.

Can you confirm that disabling the setting doesn't fix the issue?

**Reply by AskingSamagra** [reply]
It does fix the issue, and also disables the menu cursor outside of gameplay.
To clarify, menu cursor appears in menu and non-osu gameplay with that setting enabled, and doesn't appear at all when disabled.
